docs/design/021-sse-api.md

```diff
@@ -0,0 +1,55 @@
+# Introduce an SSE API module for k6
+
+|                      |                                                              |
+|:---------------------|:-------------------------------------------------------------|
+| **author**           | @phymbert                                                    |
+| **status**           | 🔧 proposal                                                  |
+| **revisions**        | [initial](https://github.com/grafana/k6/pull/3639)           |
+| **Proof of concept** | [branch](https://github.com/phymbert/k6/tree/hp/feature/sse) |
+| **references**       | [#746](https://github.com/grafana/k6/issues/746)             |
+
+## Problem definition
+
+The current version of k6 reads the full http response body before returning to the client,
+which make impossible testing [Server-Sent Event](https://fr.wikipedia.org/wiki/Server-sent_events). 
+
+We propose to introduce a new `sse` module.
+This module is intended to offer an intuitive and user-friendly API for SSE interactions within k6 scripts.
+
+## Proposed solution
+
+We suggest to implement a minimalist, experimental (`sse`) module based on the go http client.
+The new module will allow users to interact with SSE.
+The module will provide an `open` which will allow the user to pass a setup function to configure an `event` callback as it is done in the `ws` module with `message`.
+
+### Limitation
+The module will not support async io and the javascript main loop will be blocked during the http request duration.
```

Comment:
Async support is becoming more and more important, so integrating a new module without async support is questionable. This is just my personal opinion, @mstoykov  is more competent in this...

Reply:
What do you mean by asynchronous ? it's one way server to client data flow on top of HTTP 1.1 w/ single connection. The event loop is blocked while waiting for next event, but on each new event the client callback is triggered.